File: app/src/main/java/com/dammaj/app/adapter/ContentAdapter.java
package com.dammaj.app.adapter;

import android.content.Intent;
import android.net.Uri;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.bumptech.glide.Glide;
import com.bumptech.glide.load.engine.DiskCacheStrategy;
import com.dammaj.app.R;
import com.dammaj.app.model.ContentItem;

import java.util.List;

public class ContentAdapter extends RecyclerView.Adapter<ContentAdapter.ContentViewHolder> {
    private final List<ContentItem> items;

    public ContentAdapter(List<ContentItem> items) {
        this.items = items;
    }

    @NonNull
    @Override
    public ContentViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.item_content_card, parent, false);
        return new ContentViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ContentViewHolder holder, int position) {
        ContentItem item = items.get(position);
        holder.bind(item);
    }

    @Override
    public int getItemCount() {
        return items.size();
    }

    public static class ContentViewHolder extends RecyclerView.ViewHolder {
        private final ImageView imageView;
        private final TextView titleText;
        private final TextView typeText;

        public ContentViewHolder(@NonNull View itemView) {
            super(itemView);
            imageView = itemView.findViewById(R.id.image_view);
            titleText = itemView.findViewById(R.id.title_text);
            typeText = itemView.findViewById(R.id.type_text);
        }

        public void bind(ContentItem item) {
            titleText.setText(item.getTitle());
            typeText.setText(item.getType());

            if (item.getImageUrl() != null && !item.getImageUrl().isEmpty()) {
                Glide.with(itemView.getContext())
                        .load(item.getImageUrl())
                        .diskCacheStrategy(DiskCacheStrategy.ALL)
                        .centerCrop()
                        .placeholder(R.drawable.ic_article)
                        .error(R.drawable.ic_article)
                        .into(imageView);
            } else {
                imageView.setImageResource(R.drawable.ic_article);
            }

            itemView.setOnClickListener(v -> {
                if (item.getLinkUrl() != null && !item.getLinkUrl().isEmpty()) {
                    Intent intent = new Intent(Intent.ACTION_VIEW);
                    intent.setData(Uri.parse(item.getLinkUrl()));
                    itemView.getContext().startActivity(intent);
                }
            });
        }
    }
}

File: app/src/main/java/com/alilmia/app/adapter/ContentAdapter.kt
package com.alilmia.app.adapter

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.DiffUtil
import androidx.recyclerview.widget.ListAdapter
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import com.alilmia.app.R
import com.alilmia.app.databinding.ItemBookCardBinding
import com.alilmia.app.model.ContentItem

/**
 * RecyclerView adapter for displaying content items in a card layout
 * Features image loading with Glide, click handling, and efficient list updates
 */
class ContentAdapter(
    private val onItemClick: (ContentItem) -> Unit
) : ListAdapter<ContentItem, ContentAdapter.ContentViewHolder>(ContentDiffCallback()) {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ContentViewHolder {
        val binding = ItemBookCardBinding.inflate(
            LayoutInflater.from(parent.context),
            parent,
            false
        )
        return ContentViewHolder(binding, onItemClick)
    }

    override fun onBindViewHolder(holder: ContentViewHolder, position: Int) {
        holder.bind(getItem(position))
    }

    class ContentViewHolder(
        private val binding: ItemBookCardBinding,
        private val onItemClick: (ContentItem) -> Unit
    ) : RecyclerView.ViewHolder(binding.root) {

        fun bind(item: ContentItem) {
            // Load image with Glide
            if (!item.imageUrl.isNullOrEmpty()) {
                Glide.with(binding.root.context)
                    .load(item.imageUrl)
                    .placeholder(R.drawable.placeholder_book)
                    .error(R.drawable.error_image)
                    .centerCrop()
                    .into(binding.imageView)
            } else {
                binding.imageView.setImageResource(R.drawable.placeholder_book)
            }
            
            // Set title
            binding.titleText.text = item.title
            
            // Set author if available
            if (!item.author.isNullOrEmpty()) {
                binding.authorText.text = item.author
                binding.authorText.visibility = View.VISIBLE
            } else {
                binding.authorText.visibility = View.GONE
            }
            
            // Set content type indicator
            binding.typeIndicator.setImageResource(getTypeIndicator(item.type))
            
            // Set click listener
            binding.root.setOnClickListener {
                onItemClick(item)
            }
        }

        private fun getTypeIndicator(type: com.alilmia.app.model.ContentType): Int {
            return when (type) {
                com.alilmia.app.model.ContentType.BOOK -> R.drawable.ic_book
                com.alilmia.app.model.ContentType.AUDIO -> R.drawable.ic_audio
                com.alilmia.app.model.ContentType.LESSON -> R.drawable.ic_lesson
                com.alilmia.app.model.ContentType.FATWA -> R.drawable.ic_fatwa
                com.alilmia.app.model.ContentType.SCHOLAR -> R.drawable.ic_scholar
                com.alilmia.app.model.ContentType.ARTICLE -> R.drawable.ic_article
            }
        }
    }

    private class ContentDiffCallback : DiffUtil.ItemCallback<ContentItem>() {
        override fun areItemsTheSame(oldItem: ContentItem, newItem: ContentItem): Boolean {
            return oldItem.id == newItem.id
        }

        override fun areContentsTheSame(oldItem: ContentItem, newItem: ContentItem): Boolean {
            return oldItem == newItem
        }
    }
}